### PR TITLE
[SDKv2] Remove deprecated attributes from resources

### PR DESCRIPTION
**Pull request by skarimo**
This PR removes the following deprecated attributes:

Pagerduty: `"services", “individual_services"`
Users: `Removed all V1 Users code and attributes`
Synthetics: `“assertions", “variable”`
Dashboard: `“Logset”, "Count”, “Start”`

Additionally, some unused functions are removed as well. E.g. buildDatadogWidgetLayoutDeprecated,

### CODE REVIEW

File: datadog/resource_datadog_user.go [hunk 1]
```diff
@@ -2,24 +2,13 @@ package datadog
...
-	"strings"
-
-	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
-
 	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/zorkian/go-datadog-api"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+	"log"
```
Comment (zippolyte) [inline datadog/resource_datadog_user.go:9]:
```suggestion
```

File: datadog/resource_datadog_user.go [hunk 2]
```diff
@@ -2,24 +2,13 @@ package datadog
 
 import (
 	"context"
```
Comment (zippolyte) [inline datadog/resource_datadog_user.go:4]:
```suggestion
	"context"
	"log"
```

File: datadog/tests/resource_datadog_user_test.go [hunk 3]
```diff
@@ -362,43 +335,13 @@ resource "datadog_user" "bar" {
...
-	return nil
-}
-
 func datadogUserV2DestroyHelper(ctx context.Context, s *terraform.State, client *datadogV2.APIClient) error {
 	for _, r := range s.RootModule().Resources {
 		id := r.Primary.ID
 		userResponse, httpResponse, err := client.UsersApi.GetUser(ctx, id).Execute()
 
 		if err != nil {
+			fmt.Println("Is it here?", httpResponse)
```
Comment (zippolyte) [inline datadog/tests/resource_datadog_user_test.go:344]:
```suggestion
```

File: docs/resources/synthetics_test.md [hunk 4]
```diff
@@ -215,7 +214,6 @@ resource "datadog_synthetics_test" "test_browser" {
 - **step** (Block List, Deprecated) Steps for browser tests. **Deprecated.** Define `browser_step` blocks instead. (see [below for nested schema](#nestedblock--step))
```
Comment (zippolyte) [inline docs/resources/synthetics_test.md:214]: This should be removed too no ?

**Review by zippolyte — CHANGES_REQUESTED** [review]
Mostly nits

**Review by pietdaniel — APPROVED** [review]

File: docs/resources/synthetics_test.md (hunk 4 above)
Comment (skarimo) [reply docs/resources/synthetics_test.md:214]: Good catch removed it

File: datadog/resource_datadog_user.go (hunk 1 above)
Comment (skarimo) [reply datadog/resource_datadog_user.go]: The log import is used in the resource to print some helpful messages.

File: datadog/resource_datadog_user.go (hunk 2 above)
Comment (skarimo) [reply datadog/resource_datadog_user.go:4]: The log import is used in the resource to print some helpful messages.

Comment (zippolyte) [reply datadog/resource_datadog_user.go:4]: sure, but let's keep it grouped with the stdlib imports :)

**Review by zippolyte — APPROVED** [review]

**Comment by ianvernon** [comment]
I realize I'm commenting on a very old PR here, but why was `individual_services` removed as a deprecated field? It is *not* marked as deprecated in the documentation of the most recent version of v2 of the Terraform provider, and simply is gone in v3 without any clear mention of what to do. Is this an oversight? https://registry.terraform.io/providers/DataDog/datadog/2.26.1/docs/resources/integration_pagerduty

**Comment by skarimo** [comment]
@ianvernon 👋  the `individual_services` attribute was used to indicate whether user is creating services using the `datadog_integration_pagerduty_service_object` or `datadog_integration_pagerduty` resource. 

As the deprecated ability to add services using the `datadog_integration_pagerduty` resource was removed, the `individual_services` field was no longer applicable and did not have any further usage in the resource.

Please use the `datadog_integration_pagerduty_service_object` resource instead to create Pagerduty services.

**Comment by ianvernon** [comment]
@skarimo thanks for that information! Appreciate the context.

**Pull request merged**